BraveWallet/Crypto/NetworkSelectionView.swift

```diff
@@ -0,0 +1,336 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import BraveCore
+import DesignSystem
+import SwiftUI
+
+struct NetworkSelectionView: View {
+  
+  var keyringStore: KeyringStore
+  @ObservedObject var networkStore: NetworkStore
+  @ObservedObject var store: NetworkSelectionStore
+  @Environment(\.presentationMode) @Binding private var presentationMode
+  
+  init(
+    keyringStore: KeyringStore,
+    networkStore: NetworkStore,
+    networkSelectionStore: NetworkSelectionStore
+  ) {
+    self.keyringStore = keyringStore
+    self.networkStore = networkStore
+    self.store = networkSelectionStore
+  }
+  
+  var body: some View {
+    List {
+      Section {
+        ForEach(store.primaryNetworks) { presentation in
+          NetworkRowView(
+            presentation: presentation,
+            selectedNetwork: networkStore.selectedChain,
+            detailTappedHandler: {
+              store.detailNetwork = presentation
+            }
+          )
+          .onTapGesture {
+            selectNetwork(presentation.network)
+          }
```

Comment:
Shouldn't we be using a Button?

Reply:
Fixed in [c29f314](https://github.com/brave/brave-ios/pull/5706/commits/c29f314f45ae50559626c245a8c40fe2dbdfd34e)